src/reagent/impl/template.cljs

```diff
@@ -270,13 +342,20 @@
 (declare as-element)
 
 (defn native-element [parsed argv first]
-  (let [comp ($ parsed :name)]
+  (let [comp ($ parsed :name)
+        synthetic-input ($ parsed :syntheticInput)
+        synthetic-on-update ($ parsed :syntheticOnUpdate)
```

Comment:
Move `synthetic-on-update` and `synthetic-on-change` under the then branch of `if` where checking for `synthetic-input`, so that these aren't unnecessarily evaluated for each element.

Reply:
@Deraen Thanks for looking it over!

> But my point is, if I understand correctly, this change only affects `adapt-react-class`? Other cases should work already.

Yes I believe you're right, I could have been more specific, this is about `adapt-react-class`.